server/src/main/java/org/elasticsearch/internal/VersionExtension.java

```diff
@@ -18,14 +18,16 @@ public interface VersionExtension {
     /**
      * Returns the {@link TransportVersion} that Elasticsearch should use.
      * <p>
-     * This must be at least equal to the latest version found in {@link TransportVersion} V_* constants.
+     * This must be at least as high as the given fallback.
+     * @param fallback The latest transport version from server
      */
-    TransportVersion getCurrentTransportVersion();
```

Comment:
Is it worth specifying default implementations for these?

Reply:
There's only meant to be one impl, so IMO no.